src/lib/Notification/NotificationPoolAwareTrait.php

```diff
@@ -82,4 +83,16 @@ protected function addErrorNotification(string $message, array $params = [], $do
 
         $this->notificationPool->add($notification);
     }
+
+    /**
+     * Send last notification message as a response.
+     */
+    protected function notificationResponse()
```

Comment:
it's a trait the controller has no issue with it being protected, it should not be used outside of the controller